src/microbe_stage/Microbe.Contact.cs

```diff
@@ -505,80 +505,88 @@ public void Kill()
-            };
-
-            // They were added in order already so looping through this other thing is fine
-            foreach (var entry in compoundsToRelease)
-            {
-                var compoundValue = new ChunkConfiguration.ChunkCompound
+                var chunkType = new ChunkConfiguration
                 {
-                    // Randomize compound amount a bit so things "rot away"
-                    Amount = (entry.Value / random.Next(amount / 3.0f, amount)) *
-                        Constants.CORPSE_COMPOUND_COMPENSATION,
+                    ChunkScale = 1.0f,
+                    Dissolves = true,
+                    Mass = 1.0f,
+                    Radius = 1.0f,
+                    Size = 3.0f,
+                    VentAmount = 0.1f,
+
+                    // Add compounds
+                    Compounds = new Dictionary<Compound, ChunkConfiguration.ChunkCompound>(),
                 };
 
-                chunkType.Compounds[entry.Key] = compoundValue;
-            }
-
-            chunkType.Meshes = new List<ChunkConfiguration.ChunkScene>();
-
-            var sceneToUse = new ChunkConfiguration.ChunkScene();
-
-            // Try all organelles in random order and use the first one with a scene for model
-            foreach (var organelle in organelles.OrderBy(_ => random.Next()))
-            {
-                if (!string.IsNullOrEmpty(organelle.Definition.CorpseChunkScene))
+                // They were added in order already so looping through this other thing is fine
+                foreach (var entry in compoundsToRelease)
                 {
-                    sceneToUse.LoadedScene = organelle.Definition.LoadedCorpseChunkScene;
-                    break;
+                    var compoundValue = new ChunkConfiguration.ChunkCompound
+                    {
+                        // Randomize compound amount a bit so things "rot away"
+                        Amount = (entry.Value / random.Next(amount / 3.0f, amount)) *
+                            Constants.CORPSE_COMPOUND_COMPENSATION,
+                    };
+
+                    chunkType.Compounds[entry.Key] = compoundValue;
                 }
 
-                if (!string.IsNullOrEmpty(organelle.Definition.DisplayScene))
+                chunkType.Meshes = new List<ChunkConfiguration.ChunkScene>();
+
+                // Sets the default chunk to be the cytoplasm model incase a chunk can't be spawned
+                var sceneToUse = new ChunkConfiguration.ChunkScene();
+                sceneToUse.LoadedScene = SimulationParameters.Instance
+                    .GetOrganelleType(Constants.DEFAULT_CHUNK_MODEL_NAME).LoadedCorpseChunkScene;
+
+                // Will only loop if there are still organelles available
+                organellesAvailableEnumerator?.MoveNext();
+                if (i < organelles.Count && organellesAvailableEnumerator?.Current != null)
                 {
```

Comment:
There may be a better way to do this but this is the best solution I came up with